### PR TITLE
SDK-1127: Source constraints

### PR DESCRIPTION
**Pull request by MrBurtyyy**
## Added

* Support for source constraints for `WantedAttribute` when using Dynamic Scenarios

### CODE REVIEW

File: yoti-sdk-impl/src/main/java/com/yoti/api/client/shareurl/constraint/SimpleSourceConstraintBuilder.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    @Override
+    public SourceConstraintBuilder withSoftPreference(boolean softPreference) {
+        this.softPreference = softPreference;
+        return this;
+    }
+
+    @Override
+    public SourceConstraint build() {
+        return new SimpleSourceConstraint(
+                Collections.unmodifiableList(wantedAnchors),
```
Comment (echarrod) [inline yoti-sdk-impl/src/main/java/com/yoti/api/client/shareurl/constraint/SimpleSourceConstraintBuilder.java]: Not sure if we need to enforce this being unmodifiable here - this is just for the client building the request to be sent to us, I don't think we mind if it's altered before it's sent to us (unlike the profile objects that we return), would you agree?

File: yoti-sdk-impl/src/main/java/com/yoti/api/client/shareurl/policy/SimpleWantedAttribute.java [hunk 2]
```diff
@@ -17,15 +20,24 @@
...
+    @JsonProperty("constraints")
+    private final List<Constraint> constraints;
+
+    SimpleWantedAttribute(String name, String derivation, boolean optional, boolean acceptSelfAsserted, List<Constraint> constraints) {
         this.name = name;
         this.derivation = derivation;
         this.optional = optional;
+        this.acceptSelfAsserted = acceptSelfAsserted;
+
+        // TODO: Check with Mike if this is ok?
```
Comment (echarrod) [inline yoti-sdk-impl/src/main/java/com/yoti/api/client/shareurl/policy/SimpleWantedAttribute.java]: TODO

Comment (MrBurtyyy) [reply yoti-sdk-impl/src/main/java/com/yoti/api/client/shareurl/policy/SimpleWantedAttribute.java]: Fixed in https://github.com/getyoti/yoti-java-sdk/pull/138/commits/8bde085bd4cc694e43381e6a29a1ccbf62662768

File: yoti-sdk-impl/src/main/java/com/yoti/api/client/shareurl/constraint/SimpleSourceConstraintBuilder.java (hunk 1 above)
Comment (MrBurtyyy) [reply yoti-sdk-impl/src/main/java/com/yoti/api/client/shareurl/constraint/SimpleSourceConstraintBuilder.java]: As Lists are passed by reference, would prefer to keep it unmodifiable once the WantedAttribute has been built 👍

**Pull request merged**